packages/database/src/migrations/20210816012839-AddProjectPacmossi-modifies-data.js

```diff
@@ -0,0 +1,146 @@
+'use strict';
+
+import { codeToId, insertObject, generateId, findSingleRecord, deleteObject } from '../utilities';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const countryCodes = ['FJ', 'KI', 'NR', 'NU', 'PG', 'SB', 'TV', 'TO', 'VU'];
+
+const projectCode = 'pacmossi';
+const projectName = 'PacMOSSI';
+const projectDescription = 'Pacific Mosquito Surveillance Strengthening for Impact';
+
+const permissionGroups = ['PacMOSSI', 'PacMOSSI Senior'];
+
+const dashboardName = 'Vector Surveillance';
+
+const projectDashboardItemCode = 'project_details';
+const projectDashboardCode = `${projectCode}_${projectName}`;
+
+const getCountryDashboardCode = countryCode =>
+  `${countryCode}_${projectName}_${dashboardName.split(' ').join('_')}`;
+
+const hierarchyNameToId = async (db, name) => {
+  const record = await db.runSql(`SELECT id FROM entity_hierarchy WHERE name = '${name}'`);
+  return record?.rows[0].id;
+};
+
+const addCountryToProject = async (db, countryCode, entityHierarchyId) => {
+  await insertObject(db, 'entity_relation', {
+    id: generateId(),
+    parent_id: await codeToId(db, 'entity', projectCode),
+    child_id: await codeToId(db, 'entity', countryCode),
+    entity_hierarchy_id: entityHierarchyId,
+  });
+  await insertObject(db, 'dashboard', {
+    id: generateId(),
+    code: getCountryDashboardCode(countryCode),
+    name: dashboardName,
+    root_entity_code: countryCode,
+    sort_order: 1,
+  });
+};
+
+const addItemToDashboard = async (
+  db,
+  { code, dashboardCode, permissionGroup, entityTypes, projectCodes },
+) => {
+  const dashboardItemId = (await findSingleRecord(db, 'dashboard_item', { code })).id;
+  const dashboardId = (await findSingleRecord(db, 'dashboard', { code: dashboardCode })).id;
+  await insertObject(db, 'dashboard_relation', {
+    id: generateId(),
+    dashboard_id: dashboardId,
+    child_id: dashboardItemId,
+    entity_types: `{${entityTypes.join(', ')}}`,
+    project_codes: `{${projectCodes.join(', ')}}`,
+    permission_groups: `{${permissionGroup}}`,
+    sort_order: 1,
+  });
+};
+
+exports.up = async function (db) {
+  await insertObject(db, 'entity', {
+    id: generateId(),
+    code: projectCode,
+    parent_id: await codeToId(db, 'entity', 'World'),
+    name: projectName,
+    type: 'project',
+  });
+
+  const entityHierarchyId = generateId();
+  await insertObject(db, 'entity_hierarchy', {
+    id: entityHierarchyId,
+    name: projectCode,
+    canonical_types: '{country,district,field_station,larval_habitat}',
```

Comment:
@nick-beyondessential - I just saw this in another project migration (20210324041508-AddSupplyChainFijiProject-modifies-data.js). Are we meant to put canoical_types in here?

Reply:
Yes I think this makes sense as field_station & larval_habitat are not default canonical types so wouldn't be picked up otherwise. We may need to assess the hierarchy structure of each country i.e do any have sub_districts?

Reply:
I've checked it today from mock data and entities that DL has country -> district -> sub_district -> field_station hierarchy. Others are country -> district -> field_station.

Reply:
@billli0 when you're up, want to run that one past me? I.e. do I need larval_habitat, and where you checked this? :)

Reply:
@PShoe Sure, any time!

Reply:
From Edwin's [comment ](https://beyondessential.slack.com/archives/CDGS6P56K/p1627953164025600) I think we are not going to use canonical type in the future projects, as we prefer to use entity relations.

Reply:
That's right @billli0! Are `field_station` and `larval_habitat` entities all imported through the admin panel though, or are some created dynamically by filling out a survey on meditrak app? If the latter is true, we would still need canonical types for them to get automatically included in the hierarchy

Reply:
Confirmed by @gerrybes, all field_station and larval_habitat entities will be created dynamically by filling out a survey on meditrak app, so will add to canonical_types as is current code.